Library/Homebrew/cmd/update.sh

```diff
@@ -218,8 +218,8 @@ merge_or_rebase() {
   if [[ "$DIR" = "$HOMEBREW_REPOSITORY" && -n "$HOMEBREW_UPDATE_TO_TAG" ]]
   then
     UPSTREAM_TAG="$(git tag --list |
-                    sort --field-separator=. --key=1,1nr -k 2,2nr -k 3,3nr |
-                    grep --max-count=1 '^[0-9]*\.[0-9]*\.[0-9]*$')"
+                    sort --field-separator=. -k 1,1nr -k 2,2nr -k 3,3nr -k 4,4nr |
+                    grep -E --max-count=1 '^[0-9]+\.[0-9]+\.[0-9]+(\.[0-9]+)?$')"
```

Comment:
It's standardized that a missing field sorts as the empty string.
http://pubs.opengroup.org/onlinepubs/9699919799/utilities/sort.html
> The notation:
`-k field_start[type][,field_end[type]]`
shall define a key field that begins at field_start and ends at field_end inclusive, unless field_start falls beyond the end of the line or after field_end, in which case the key field is empty.